File: backend/src/main/java/com/ttak/backend/domain/history/repository/HistoryRepository.java
package com.ttak.backend.domain.history.repository;

import org.springframework.data.jpa.repository.JpaRepository;

import com.ttak.backend.domain.history.entity.History;
import com.ttak.backend.domain.history.repository.customRepository.HistoryCustomRepository;

public interface HistoryRepository extends JpaRepository<History, Long>, HistoryCustomRepository {
}

File: ttak/app/src/main/java/com/ttak/android/data/repository/history/HistoryRepository.kt
package com.ttak.android.data.repository.history

import com.ttak.android.domain.model.history.HistoryInfo

interface HistoryRepository {

    suspend fun getWeeklyPickCount(): Int
    suspend fun getWeeklyWatchingCount(): Int
    suspend fun getMessages(): List<HistoryInfo>
    suspend fun sendMessage(message: String)
}